File: python_codewars/12.py
def pascal(p):
    #your code here
    if p==1:
        return [[1]];
    elif p==2:
        return [[1],[1,2]];
    ret = [[1],[1,1]];
    for i in range(1, p-1):
        #print(i);
        re=[1];
        for j in range(i):
            #print(j);
            re.append(ret[i][j]+ret[i][j+1]);
        re.append(1);
        ret.append(re);
        #print(ret);
    return ret;

pascal(1);
pascal(3);
pascal(5);

File: hackerrank/12.cpp
/*
Fibonacci Modified

Sample Input

0 1 5
Sample Output

5
*/
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int a, b, n;
    cin>>a>>b>>n;
    vector<long long> f(n);
    f[0] = a;
    f[1] = b;
    for(int i=2; i<n; i++)
        f[i]=f[i-1]*f[i-1] + f[i-2];
    cout<<f[n-1]<<endl;
    return 0;
}
